app/views/components/_autocomplete.html.erb

```diff
@@ -0,0 +1,41 @@
+<%
+  id ||= "autocomplete-#{SecureRandom.hex(4)}"
+  name ||= "autocomplete"
+  label ||= nil
+  value ||= nil
+  options ||= []
+
+  error_id = "error-#{SecureRandom.hex(4)}"
+  error_items ||= []
+
+  classes = %w(app-c-autocomplete govuk-form-group)
+  classes << "govuk-form-group--error" if error_items.any?
+%>
+<% if label && options.any? %>
```

Comment:
Shouldn't we error if these aren't provided. It feels like unexpected behaviour for the component to quietly do nothing when the wrong or missing inputs are given (I'm getting flash backs to the incident behind https://github.com/alphagov/govuk_publishing_components/pull/2335)

Reply:
I think most of our components fail quietly if the right data isn't passed to them? Do you have an example of component that doesn't?

Reply:
@kevindew I've found an example and updated this PR so that the autocomplete raises an error if the right things aren't passed to it - see the extra commit (will squash once approved).

Reply:
Super. It's a shame that this is a pattern that has crept into the components, we should try reverse that.

Reply:
One to consider definitely. It looks as though that change to the image card has been reverted, although I don't remember the exact reasons behind it.